image/image.go

```diff
@@ -8,10 +8,24 @@ import (
 // Type must be a "Backing file format name string" that appears in QCOW2.
 type Type string
 
+type Extent struct {
+	// Offset from start of the image in bytes.
+	Start int64 `json:"start"`
+	// Length of this extent in bytes.
+	Length int64 `json:"length"`
+	// Set if this extent is allocated.
+	Allocated bool `json:"allocated"`
+	// Set if this extent is read as zeros.
+	Zero bool `json:"zero"`
+	// Set if this extent is compressed.
+	Compressed bool `json:"compressed"`
+}
+
 // Image implements [io.ReaderAt] and [io.Closer].
 type Image interface {
 	io.ReaderAt
 	io.Closer
+	Next(start, length int64) (Extent, error)
```

Comment:
Does this return EOF on the end of the image?

Reply:
No, since you know the size of the image and must use the correct size to use this. If you specify a range which is after the end of the image it will fail.